src/v/raft/consensus.cc

```diff
@@ -1227,6 +1227,29 @@ ss::future<vote_reply> consensus::do_vote(vote_request&& r) {
         reply.granted = false;
         return ss::make_ready_future<vote_reply>(reply);
     }
+
+    // Optimization: for vote requests from nodes that are likely
+    // to have been recently restarted (have failed heartbeats
+    // and an <= present term), reset their RPC backoff to get
+    // a heartbeat sent out sooner.
+    if (is_leader() and r.term <= _term) {
+        // Look up follower stats for the requester
+        if (auto it = _fstats.find(r.node_id); it != _fstats.end()) {
+            auto& fstats = it->second;
+            if (fstats.heartbeats_failed) {
+                vlog(
+                  _ctxlog.debug,
+                  "Vote request from peer {} with heartbeat failures, "
+                  "resetting backoff",
+                  r.node_id);
+                return _client_protocol.reset_backoff(r.node_id.id())
```

Comment:
What if the node is not restarted but just unresponsive?

Reply:
I think the same logic should apply in both cases: the vote/prevote message is a signal that the node is alive, and therefore we shouldn't be backing off from sending it heartbeats.